src/Extensions/ServiceFabric.Abstractions/ServiceFabricMachineInformation.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Fabric;
+using System.Net;
+using Microsoft.Extensions.Options;
+using Microsoft.Omex.Extensions.Abstractions;
+
+namespace Microsoft.Omex.Extensions.ServiceFabric.Abstractions
+{
+	internal class ServiceFabricMachineInformation : EmptyMachineInformation
+	{
+		public ServiceFabricMachineInformation(ServiceContext context, IOptions<MachineInformationSettings> settings)
+		{
+			MachineName = GetMachineName();
+			DeploymentSlice = DefaultEmptyValue;
+			IsCanary = false;
+			MachineCount = 1;
+
+			ServiceName = context.ServiceName.ToString();
+			ICodePackageActivationContext activationContext = context.CodePackageActivationContext ?? FabricRuntime.GetActivationContext();
+			MachineRole = activationContext.ApplicationName ?? DefaultEmptyValue;
+			BuildVersion = activationContext.CodePackageVersion;
+
+			NodeContext nodeContext = context.NodeContext ?? FabricRuntime.GetNodeContext();
+			MachineId = $"{MachineName}_{nodeContext.NodeName}";
```

Comment:
>$"{MachineName}_{nodeContext.NodeName}"; [](start = 15, length = 40)

Enclose this within `FormattableString.Invariant`. Otherwise, this can vary between different cultures

Reply:
Done, thank you, could you expand on what could be different here? Is not it just string concatenation

Reply:
It'll almost certainly stay the same in reality, but the standard is I believe that this is open to change as necessary. For example, it would be possible for the framework to insert RTL or LTR directionality markers into the string for certain markets if it wished. It's safest to include interpolated strings within FormattableString.Invariant to resolve this.

---
In reply to: [366261897](https://github.com/microsoft/Omex/pull/153#discussion_r366261897) [](ancestors = 366261897)